src/shimmerHttp.js

```diff
@@ -0,0 +1,215 @@
+import http from 'http';
+import https from 'https';
+import url from 'url';
+import { debuglog } from 'util';
+import shimmer from 'shimmer';
+import Perf from 'performance-node';
+import uuid from 'uuid/v4';
+import pickBy from 'lodash.pickby';
+import isArray from 'isarray';
+import { flatten } from 'flat';
+
+const debug = debuglog('@iopipe/trace');
+
+/*eslint-disable babel/no-invalid-this*/
+
+function unwrap() {
+  if (http.get.__wrapped) {
```

Comment:
This could all be reduced to a loop over an array of `['get', 'request', ...']`.

Reply:
Yes